capstone/src/main/java/com/google/sps/servlets/ExploreServlet.java

```diff
@@ -48,6 +54,18 @@ private Club createClubFromEntity(Entity entity) {
         ServletUtil.getPropertyList(entity, Constants.OFFICER_PROP),
         entity.getProperty(Constants.DESCRIP_PROP).toString(),
         entity.getProperty(Constants.WEBSITE_PROP).toString(),
-        key);
+        key,
+        Long.parseLong(entity.getProperty(Constants.TIME_PROP).toString()));
+  }
+
+  private Comparator<Club> getComparator(String sort) {
+    if (sort.equals(Constants.ALPHA_SORT_PROP)) {
+      return Comparator.comparing(
+          club -> club.getName().toLowerCase()); // Should be case-insensitive
+    } else if (sort.equals(Constants.SIZE_SORT_PROP)) {
+      return Collections.reverseOrder(Comparator.comparing(club -> club.getSize()));
+    } else {
+      return Comparator.comparing(club -> club.getCreationTime());
+    }
```

Comment:
nit: you can also use switch statements which are a bit more scalable than using multiple if-else statements, especially if we add more sort options in the future

https://www.w3schools.com/java/java_switch.asp

Reply:
Cool! For some reason, I thought Strings were the one thing you couldn't switch in Java